### ISSUE TITLE
"Create new project" wizard inside LSP

### ISSUE THREAD

**Issue by okmaybenot** [clojure-lsp/clojure-lsp#1918]
**Is your feature request related to a problem? Please describe.**
Languages ​​and ecosystems often bundle options for creating a new project through a wizard, making it easier to sample and test things. For example, see the screenshot for Flutter inside VSCode

**Describe the solution you'd like**
I propose to extend LSP to create something similar for Clojure and all editors that use LSP.

**Describe alternatives you've considered**
Current users need to switch between terminal and VSCode to create new projects, which is cumbersome.

![image](https://github.com/user-attachments/assets/42e0ec10-f6da-4a7e-878c-e345d57472f1)

**Comment by ericdallo** [comment]
Yeah, I implemented this support manually for clojure-lsp-intellij but I can see how this is convenient for other editors and have in a common place

**Comment by okmaybenot** [comment]
Sure, the flow is simple, but convenient – select a template, provide a name, then the basic project is created and opened, see the attached video.

https://github.com/user-attachments/assets/26781167-9573-4b34-9d7c-50ef82aef646

I believe, given the Clojure ecosystem versatility, the key part to good UX is to provide extension point for the templates. 

For example, [ClojureDart](https://github.com/Tensegritics/ClojureDart?tab=readme-ov-file#your-first-app) relies on manual project creation via the command line, which can be tedious, especially when you try a lot of things. I tried to simplify this by replicating a Flutter-like experience with [Neil](https://github.com/babashka/neil) (which is based on [deps.new](https://github.com/seancorfield/deps-new)) and it was not easy at all, so I gave up after an hour or so.
